tantar/src/main/java/org/opencadc/tantar/policy/InventoryIsAlwaysRight.java

```diff
@@ -95,25 +95,31 @@ public void resolve(final Artifact artifact, final StorageMetadata storageMetada
             // happen in the case where all files are managed by the inventory but an intervention outside of the
             // Storage Inventory caused a file to disappear.  The file may not have been fully uploaded to begin with
             // either.
-            reporter.report(String.format("Resetting Artifact %s as per policy.", artifact.storageLocation));
+            reporter.report("Resetting Artifact " + artifact.storageLocation + " as per policy.");
 
             validateEventListener.markAsNew(artifact);
         } else if (artifact == null) {
-            reporter.report(String.format("Removing Unknown File %s as per policy.",
-                                          storageMetadata.getStorageLocation()));
+            reporter.report("Removing Unknown File " + storageMetadata.getStorageLocation() + " as per policy.");
 
             validateEventListener.delete(storageMetadata);
         } else {
             // Check metadata for discrepancies.
             if (haveDifferentStructure(artifact, storageMetadata)) {
-                // Then prefer the Artifact.
-                reporter.report(String.format("Replacing File %s as per policy.",
-                                              storageMetadata.getStorageLocation()));
+                // The metadata differs, but for valid reasons (update).
+                if (storageMetadata.isValid()) {
```

Comment:
I feel like the isValid check should be in the outer sequence of if/else and not inside the haveDifferentStructure because it is more fundamental (not valid is not equivalent to storageMetadata == null but closer to that than just "mismatched metadata")... basically, if not valid, the metadata comparison in haveDifferentStructure should not even be performed.

Also, if storageMetadata.isValid() == false, you do need to both `delete(storageMetadata)` and `markAsNew(artifact)`

Aside; use of the word "corrupt" here is probably overstating things most of the time and is likely to make ops people panic... they will interpret "files are being corrupted" to mean something else

Reply:
Thanks.  I've added some rework for messages and the `isValid` check.  I've modified the message to remove `Corrupt` for the `StorageIsAlwaysRight` policy, but corruption _may_ actually be the cause.